### PR TITLE
[WIP] fix: repro ingress-nginx replica latency

### PR DESCRIPTION
**Pull request by jackfrancis**
**Reason for Change**:

This PR introduces additional E2E coverage to attempt to repro https://github.com/Azure/aks-engine/issues/3392

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: test/e2e/kubernetes/deployment/deployment.go [hunk 1]
```diff
@@ -123,7 +123,7 @@ metadata:
     role: %s
   name: %s
 spec:
-  replicas: 1
+  replicas: 9
```
Comment (jackfrancis) [inline test/e2e/kubernetes/deployment/deployment.go:126]: artificially setting replica count high here

File: test/e2e/kubernetes/kubernetes_test.go [hunk 2]
```diff
@@ -1422,8 +1422,12 @@ var _ = Describe("Azure Container Cluster using the Kubernetes Orchestrator", fu
 				Expect(err).NotTo(HaveOccurred())
 
 				By("Ensuring we can create a curl pod to connect to the service")
-				ilbCurlPod, err := pod.RunLinuxWithRetry("byrnedo/alpine-curl", "curl-to-ilb", "default", fmt.Sprintf("curl %s", sILB.Status.LoadBalancer.Ingress[0]["ip"]), false, 1*time.Minute, cfg.Timeout)
-				Expect(err).NotTo(HaveOccurred())
+				for i := 0; i < 100; i++ {
+					ilbCurlPod, err := pod.RunLinuxWithRetry("byrnedo/alpine-curl", "curl-to-ilb", "default", fmt.Sprintf("curl %s", sILB.Status.LoadBalancer.Ingress[0]["ip"]), false, 1*time.Minute, 1*time.Second, 30*time.Second)
```
Comment (jackfrancis) [inline test/e2e/kubernetes/kubernetes_test.go:1426]: this test (which we run 100 times in succession) will fail if it takes longer than one second to curl to the LB IP

File: test/e2e/kubernetes/deployment/deployment.go (hunk 1 above)
Comment (gaieges) [reply test/e2e/kubernetes/deployment/deployment.go:126]: Is this what gets used to set up the nginx pod count?  I presume our issue was something to do with the azure LB not being aware of what nodes the nginx pods are living on.

Comment (jackfrancis) [reply test/e2e/kubernetes/deployment/deployment.go:126]: Correct. These are the LB specs we test against the ingress-nginx deployment (w/ 9 replicas in this test):

https://github.com/Azure/aks-engine/blob/master/test/e2e/kubernetes/workloads/ingress-nginx-ilb.yaml
https://github.com/Azure/aks-engine/blob/master/test/e2e/kubernetes/workloads/ingress-nginx-elb.yaml

As you can see, the LB routes to the pod (and eventually the node the pod is scheduled onto) via the app selector.

What's different about your scenario?

**Comment by posix4e** [comment]
Neat;

File: test/e2e/kubernetes/deployment/deployment.go (hunk 1 above)
Comment (posix4e) [reply test/e2e/kubernetes/deployment/deployment.go:126]: can we try this with k8s 1.16?

Comment (jackfrancis) [reply test/e2e/kubernetes/deployment/deployment.go:126]: Updated CI config, the next run will test 1.15 and 1.16

**Pull request closed without merging**